client-dynamodb/src/main/kotlin/app/cash/backfila/client/dynamodb/DynamoDbBackfill.kt

```diff
@@ -76,4 +76,7 @@ abstract class DynamoDbBackfill<I : Any, P : Any> : Backfill {
 
   /** See [ScanRequest.setExpressionAttributeNames]. */
   open fun expressionAttributeNames(config: BackfillConfig<P>): Map<String, String>? = null
+
+  /** See [ScanRequest.setIndexName]. */
+  open fun indexName(config: BackfillConfig<P>): String? = null
```

Comment:
I wonder if I should encapsulate all these arguments? Builder method might be nice and clear too.